.github/workflows/build.yml

```diff
@@ -0,0 +1,82 @@
+name: Build
+
+on:
+  workflow_call:
+    inputs:
+      tag_name:
+        description: "Tag name"
+        required: true
+        type: string
```

Comment:
WDYT of having a boolean input for `upload_artifacts`? So that we skip that for the non-publish case?